### PR TITLE
#93 Implement endpoints for a user to delete a dossier.

### PR DESCRIPTION
**Pull request by zihao429**
Setup relevant endpoints for dossier deletion. Relates to #93

### CODE REVIEW

File: Server/ConcordiaCurriculumManager/Controllers/DossierController.cs [hunk 1]
```diff
@@ -152,4 +152,37 @@ public async Task<ActionResult> CreateDossierForUser([FromBody, Required] Create
...
+    [SwaggerResponse(StatusCodes.Status201Created, "Dossier deleted successfully", typeof(DossierDTO))]
+    public async Task<ActionResult> DeleteDossier([FromBody, Required] DeleteDossierDTO dossier)
+    {
+        try
+        {
+            var user = await _userService.GetCurrentUser();
+            var deletedDossier = await _dossierService.DeleteDossier(dossier, user);
+            var deletedDossierDTO = _mapper.Map<DossierDTO>(deletedDossier);
+
+            return Created($"/{nameof(DeleteDossier)}", deletedDossierDTO);
```
Comment (Ghaith-Chrit) [inline Server/ConcordiaCurriculumManager/Controllers/DossierController.cs:169]: Normally an HTTP Delete returns a 204 (No Content) status code if the action has been enacted

File: Server/ConcordiaCurriculumManager/Services/DossierService.cs [hunk 2]
```diff
@@ -81,6 +82,32 @@ public async Task<Dossier> EditDossier(EditDossierDTO d, User user)
         return dossier;
     }
 
+
+    public async Task<Dossier> DeleteDossier(DeleteDossierDTO d, User user)
```
Comment (Ghaith-Chrit) [inline Server/ConcordiaCurriculumManager/Services/DossierService.cs:86]: This method does not need to return anything. Unless there is a use case that I am missing.

**Review by Ghaith-Chrit — CHANGES_REQUESTED** [review]
Looks good just a couple of small stuff. Also can you please write some tests for this functionality?

File: Server/ConcordiaCurriculumManager/Controllers/DossierController.cs (hunk 1 above)
Comment (zihao429) [reply Server/ConcordiaCurriculumManager/Controllers/DossierController.cs]: Yes, you are right. Should be good now

File: Server/ConcordiaCurriculumManager/Services/DossierService.cs (hunk 2 above)
Comment (zihao429) [reply Server/ConcordiaCurriculumManager/Services/DossierService.cs]: Sounds good

**Comment by zihao429** [comment]
> Looks good just a couple of small stuff. Also can you please write some tests for this functionality?

Kenny is going to do the testing.

**Review by Ghaith-Chrit — APPROVED** [review]

**Pull request merged**